core/src/main/java/io/grpc/EquivalentAddressGroup.java

```diff
@@ -60,6 +61,33 @@ public EquivalentAddressGroup(SocketAddress addr) {
   }
 
   /**
+   * Converts list of ResolvedServerInfoGroup objects into one EquivalentAddressGroup object.
+   */
+  public static EquivalentAddressGroup fromResolvedServerInfoGroups(
+      List<ResolvedServerInfoGroup> groupList) {
+    List<SocketAddress> addrs = new ArrayList<SocketAddress>(groupList.size());
+    for (ResolvedServerInfoGroup group : groupList) {
+      for (ResolvedServerInfo srv : group.getResolvedServerInfoList()) {
+        addrs.add(srv.getAddress());
+      }
+    }
+    return new EquivalentAddressGroup(addrs);
+  }
+
+  /**
+   * Converts list of ResolvedServerInfoGroup objects into list of EquivalentAddressGroup objects.
+   */
+  // TODO(lukaszx0) since it's returning collection, it might not belong here?
+  public static List<EquivalentAddressGroup> fromResolvedServerInfoGroupsAsList(
```

Comment:
No we are not. Even with Java pre-8, it only adds some convenience, but it will be much used. Eventually Java 8 will predominate and this helper method will become obsolete. I'd rather remove this method for a smaller API surface.